src/AstArgument.h

```diff
@@ -258,9 +258,16 @@ class AstTerm : public AstArgument {
  */
 
 class AstFunctor : public AstTerm {
+public:
+    virtual bool isOverloaded() const {
```

Comment:
Can we remove this method from AstFunctor? We don't like methods in the class hierarchy that goes beyond the description of an Ast element. If they provide any semantic information, we would prefer to place them in their own analysis class/util class (depending on whether they need state).